test/api_bridge_test.rb

```diff
@@ -131,27 +133,32 @@ def before_setup
     # calculate the start when something IS sent in about it
     assert_equal 8, @api.prepare_options({ "num" => "2", "page" => "5" })["start"]
     assert_equal 20, @api.prepare_options({ "num" => "20", "page" => "2" })["start"]
+  end
 
-    # check that there is a filter still applied in the api settings
-    assert_equal ["collection|letters"], opts["f"]
+  test "prepare_options f[]" do
+    # check that there is a filter applied in the app wide settings
+    assert_equal @f_app, @api.prepare_options({})["f"]
 
     # set a new filter which is not the same as the app wide filter
-    assert_equal ["collection|letters", "subcategory|memos"], @api.prepare_options({ "f" => ["subcategory|memos"] })["f"]
-  end
+    assert_equal ["collection|letters", "keyword|production", "subcategory|memos"], @api.prepare_options({ "f" => ["subcategory|memos"] })["f"]
 
-  test "prepare_options f[]" do
-    # TODO this fails to override an app wide filter, is this expected behavior?
-    # actually returns ["collection|letters", "collection|whitman", "subcategory|marginalia"]
-    setting = ["collection|whitman", "subcategory|marginalia"]
-    assert_equal setting, @api.prepare_options({ "f" => setting })["f"]
+    # if a filter collides (collection in this case), make sure the newly requested one wins out!
+    assert_equal(
+      ["keyword|production", "collection|whitman", "subcategory|marginalia"],
+      @api.prepare_options({ "f" => ["collection|whitman", "subcategory|marginalia"] })["f"]
+    )
+
+    # check that filter collisions where the request has multiple of the same filter preferences request
```

Comment:
I think this and the test above are redundant. They are both just testing whether `keyword|production` is removed, right? Whether the replacement has more than one `keyword` or not doesn't matter to the code being tested from how I read it.

Reply:
I mostly wanted to make sure that both of the new keywords would make it through, partly because originally I had written the code in query.rb differently. I think I will remove the more simplistic test rather than this one.